packages/contracts/contracts/votings/majority-voting/ERC20Voting.sol

```diff
@@ -0,0 +1,157 @@
+/*
+ * SPDX-License-Identifier:    MIT
+ */
+
+pragma solidity 0.8.10;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20VotesUpgradeable.sol";
+
+import "./MajorityVoting.sol";
+
+contract ERC20Voting is MajorityVoting {
+
+    ERC20VotesUpgradeable public token;
+
+    mapping(uint256 => uint64) snapshotBlocks;
+
+    function getSnapshotBlock(uint256 _voteId) public view returns (uint64) {
```

Comment:
do we need this method if we anyways can call: ``contract.snapshotBlocks(voteId)``